### PR TITLE
introduce UnsupportedPlatformException to find interface

### PR DESCRIPTION
**Pull request by Cokemonkey11**
This allows a consuming library to distinguish a NotFound from an UnsupportedPlatform error

In particular I'll use this as a follow-up for https://github.com/wurstscript/WurstScript/pull/1075

### CODE REVIEW

File: src/main/java/net/moonlightflower/wc3libs/port/Context.java [hunk 1]
```diff
@@ -15,14 +15,15 @@
 import java.util.Stack;
 
 public class Context {
-    private static Map<Class<?>, Stack<Object>> _services = new LinkedHashMap<>();
+    private static final Map<Class<?>, Stack<Object>> _services = new LinkedHashMap<>();
```
Comment (Cokemonkey11) [inline src/main/java/net/moonlightflower/wc3libs/port/Context.java:18]: Drive-by as recommended by IDEA

File: src/main/java/net/moonlightflower/wc3libs/port/Context.java [hunk 2]
```diff
@@ -15,14 +15,15 @@
...
 
     private static Stack<Object> getServiceStack(@Nonnull Class<?> clazz) {
         if (!_services.containsKey(clazz)) _services.put(clazz, new Stack<>());
 
         return _services.get(clazz);
     }
 
+    @SuppressWarnings("unchecked")
```
Comment (Cokemonkey11) [inline src/main/java/net/moonlightflower/wc3libs/port/Context.java:26]: Drive-by as recommended by IDEA

File: src/main/java/net/moonlightflower/wc3libs/port/StdGameExeFinder.java [hunk 3]
```diff
@@ -38,7 +38,7 @@ public File find() throws NotFoundException {
                 throw e;
             }
         } else {
-            throw new NotFoundException(new Exception("system not supported: " + Orient.getSystem()));
+            throw new UnsupportedPlatformException("system not supported: " + Orient.getSystem());
```
Comment (Cokemonkey11) [inline src/main/java/net/moonlightflower/wc3libs/port/StdGameExeFinder.java:41]: (This is the key exception instance that I actually want to use in wurstscript)

File: src/main/java/net/moonlightflower/wc3libs/port/mac/MacMapsDirFinder.java
Comment (Cokemonkey11) [inline src/main/java/net/moonlightflower/wc3libs/port/mac/MacMapsDirFinder.java:1]: Note that this and the other platform-specific finders just swallow unsupportedplatformexceptions since they don't make sense

File: src/test/java/wc3libs/misc/ObjMergerTest.java [hunk 4]
```diff
@@ -15,6 +16,7 @@
 public class ObjMergerTest {
     private static final Logger log = LoggerFactory.getLogger(ObjMergerTest.class.getName());
 
+    @SuppressWarnings("DataFlowIssue")
```
Comment (Cokemonkey11) [inline src/test/java/wc3libs/misc/ObjMergerTest.java:19]: Drive-by as recommended by IDEA

File: src/test/java/wc3libs/misc/StateTest.java [hunk 5]
```diff
@@ -9,6 +9,7 @@
 import java.util.Collection;
 
 public class StateTest {
+    @SuppressWarnings({"unchecked", "rawtypes"})
```
Comment (Cokemonkey11) [inline src/test/java/wc3libs/misc/StateTest.java:12]: Drive-by as recommended by IDEA

**Comment by WaterKnight** [comment]
This changes the signature of the Finder interface, which is not really about the platform. Maybe it would be better to make it a Cause of the NotFoundException, unchecked or a subtype of it?

**Comment by Cokemonkey11** [comment]
@WaterKnight thanks for the review. I'm happy to make it unchecked

File: src/main/java/net/moonlightflower/wc3libs/port/StdGameExeFinder.java [hunk 6]
```diff
@@ -24,21 +24,13 @@ public File find() throws NotFoundException {
         if (Orient.isMacSystem()) {
             GameExeFinder macGameExeFinder = getMacGameExeFinder();
 
-            try {
-                return macGameExeFinder.get();
-            } catch (NotFoundException e) {
-                throw e;
-            }
+            return macGameExeFinder.get();
```
Comment (Cokemonkey11) [inline src/main/java/net/moonlightflower/wc3libs/port/StdGameExeFinder.java:27]: Drive-by cleanup
